WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.java

```diff
@@ -1371,6 +1382,44 @@ public boolean onImageUrlClick(String imageUrl, View view, int x, int y) {
         return showPhotoViewer(imageUrl, view, x, y);
     }
 
+    @Override
+    public boolean onFileDownloadClick(String fileUrl) {
+        FragmentActivity activity = getActivity();
+        if (activity != null
+            && fileUrl != null
+            && PermissionUtils.checkAndRequestStoragePermission(this, FILE_DOWNLOAD_PERMISSION_REQUEST_CODE)) {
+            downloadFile(fileUrl, activity);
+            return true;
+        } else {
+            mFileForDownload = fileUrl;
+            return false;
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions,
+                                           @NonNull int[] grantResults) {
+        FragmentActivity activity = getActivity();
+        if (activity != null
+            && requestCode == FILE_DOWNLOAD_PERMISSION_REQUEST_CODE
+            && (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED)) {
```

Comment:
Same as previous comment; maybe can make sense to reuse the `WPPermissionUtils.setPermissionListAsked` as for example in [here](https://github.com/wordpress-mobile/WordPress-Android/blob/0b681e358496663bce0ed4a4eb133c2738fdbe37/WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java#L525)?